test/unit/storage/FileResourceStore.test.ts

```diff
@@ -0,0 +1,225 @@
+import arrayifyStream from 'arrayify-stream';
+import { BinaryRepresentation } from '../../../src/ldp/representation/BinaryRepresentation';
+import { FileResourceStore } from '../../../src/storage/FileResourceStore';
+import fs from 'fs';
+import { LINK_TYPE_LDP_BC } from '../../../src/util/LinkTypes';
+import { MethodNotAllowedHttpError } from '../../../src/util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../../../src/util/errors/NotFoundHttpError';
+import { Readable } from 'stream';
+import { RepresentationMetadata } from '../../../src/ldp/representation/RepresentationMetadata';
+import streamifyArray from 'streamify-array';
+import { UnsupportedMediaTypeHttpError } from '../../../src/util/errors/UnsupportedMediaTypeHttpError';
+import { DATA_TYPE_BINARY, DATA_TYPE_QUAD } from '../../../src/util/ContentTypes';
+import { namedNode, triple } from '@rdfjs/data-model';
+
+const base = 'http://test.com/';
+const root = '/Users/default/home/public/';
+
+describe('A FileResourceStore', (): void => {
+  let store: FileResourceStore;
+  let representation: BinaryRepresentation;
+  let stats: fs.Stats;
+  const rawData = 'lorem ipsum dolor sit amet consectetur adipiscing';
+
+  beforeEach(async(): Promise<void> => {
+    store = new FileResourceStore(base, root);
+
+    representation = {
+      data: streamifyArray([ rawData ]),
+      dataType: DATA_TYPE_BINARY,
+      metadata: {} as RepresentationMetadata,
+    };
+
+    stats = {
+      atime: new Date(),
+      atimeMs: 0,
+      birthtime: new Date(),
+      birthtimeMs: 0,
+      blksize: 0,
+      blocks: 0,
+      ctime: new Date(),
+      ctimeMs: 0,
+      dev: 0,
+      gid: 0,
+      ino: 0,
+      isBlockDevice(): boolean {
+        return false;
+      },
+      isCharacterDevice(): boolean {
+        return false;
+      },
+      isDirectory(): boolean {
+        return false;
+      },
+      isFIFO(): boolean {
+        return false;
+      },
+      isFile(): boolean {
+        return false;
+      },
+      isSocket(): boolean {
+        return false;
+      },
+      isSymbolicLink(): boolean {
+        return false;
+      },
+      mode: 0,
+      mtime: new Date(),
+      mtimeMs: 0,
+      nlink: 0,
+      rdev: 0,
+      size: 0,
+      uid: 0,
+    };
+  });
```

Comment:
Probably can skip some of those? Only implement what is needed and cast. (Alternatively, get a real `stats` object from somewhere and overwrite the needed fields.)